libminifi/src/utils/file/FileUtils.cpp

```diff
@@ -66,21 +66,21 @@ bool contains(const std::filesystem::path& file_path, std::string_view text_to_s
   return std::search(view.begin(), view.end(), searcher) != view.end();
 }
 
-time_t to_time_t(std::filesystem::file_time_type file_time) {
-#if defined(_LIBCPP_VERSION) && _LIBCPP_VERSION < 14000
-  return std::chrono::file_clock::to_time_t(file_time);
+std::chrono::system_clock::time_point to_sys(std::filesystem::file_time_type file_time) {
+  using namespace std::chrono;  // NOLINT(build/namespaces)
+#if defined(WIN32) || (defined(_LIBCPP_VERSION) && (_LIBCPP_VERSION < 14000))
+  return system_clock::now() + duration_cast<system_clock::duration>(file_time - file_clock::now());
```

Comment:
I think this causes occasional unit test failures because we read `now()` twice, and sometimes we get two different ticks back.

As an alternative, this worked for me on Windows:
```c++
  // to_sys
  return utc_clock::to_sys(file_clock::to_utc(file_time));

  // from_sys
  return file_clock::from_utc(utc_clock::from_sys(sys_time));
```

I'm not sure if it works on libc++ version < 14, too, but I'm also not sure if we really need to support libc++ version < 14.

Reply:
It seems this method produces the same error which I encountered when I was using clock_cast (clock cast probably does this under the hood) (it loads timezonedb which based on the comments fails on older windowses https://github.com/microsoft/STL/issues/1911)
Also the macOS CI does seem to use libc++<14 (altough I might be able to update that)

Reply:
More info on the windows issue: https://github.com/microsoft/STL/issues/2446

Reply:
To support older windows it seems we would have to use the double now method or resort to hardcoded magic numbers like the file epoch is 1601.01.01. Whats your opinion?

Reply:
I see, this is unfortunate.

I think hard-coding the epoch differences (`std::filesystem::__std_fs_file_time_epoch_adjustment` in the case of Windows, and `0` in the case of `libc++ < 14`) is the least bad option.